### ISSUE TITLE
Source Code

### ISSUE THREAD

**Issue by fbnlrz** [GeekMagicClock/smalltv-ultra#40]
Hi :) First of all i love this cute little Device!

But could you Pleeeease provide the Sourcecode? 
A lot of people would love to integrate it into Home Assistant or do cool stuff with it.
Thank you!!

**Comment by Ashwagada** [comment]
Good idea.

**Comment by vjhoming** [comment]
Yes please !

**Comment by MaximNedavniy** [comment]
Source code is not only customization, but also security (it is possible to see what a device connected to the network is actually doing).

**Comment by BLKFSH** [comment]
Agreed! I'd love adding some custom features and while I could just wipe it and make my own firmware, it would be cool to have the ability to keep the current one :)

**Comment by JoeB2** [comment]
I am confident the the +1hr time bug and the the DST flag inop bug could be quickly found and corrected by many people in this community.  i.e. If only we had source code access.  Please consider providing us with source code access.

**Comment by taborrobak** [comment]
+1
a) Can't keep this on my network until I know what's going on under the hood
b) Might as well make it opensource so the community can develop it

**Comment by pakair** [comment]
source code would be nice ... 
I have an esp32 and a 3.2 inch SPI display (ILI9341) and I'd like to make the code work on a bigger display.

**Comment by Luanvbn** [comment]
Please make the source code available, I would also like to customize it with my Alexa and other things.

**Comment by ferenczy** [comment]
I'll just add that by opening the source code, the device manufacturer would also benefit because they will get a **"community developer" working 24/7 on new features, improvements, and bug fixes for free, making the device more attractive to potential buyers**. If it would attract a reasonably sized community, it may turn the device into a multi-purpose one, that may be used for all kinds of purposes. In contrast to the current state, which is cool, don't take me wrong, but it's just a simple clock with weather information, so nothing particularly interesting. The value is in the hardware, so let's take it to another level by inflating the value of the software component. And as a consequence, boost the device sales! What do you think, @GeekMagicClock, please?

**Comment by ferenczy** [comment]
Imagine creating a simple app store, where you can install various apps/plugins with a mouse click. Just by installing an additional app/plugin, the device would turn into a little Home Assistant dashboard. Or display cryptocurrency/fiat exchange rates. Or show a subscriber counter from social platforms. Basically, similar to how the LaMetric device works, for example.

**Comment by rob040** [comment]
+1
This is indeed a great device for adding features by open source community , although the PRO (ESP32) version would be better suited for adding 'apps' etc. It has more RAM and can be had with more flash. Also, the core development of ESP8266 is stopped for many years.

Similar open-source requests are in issue #56, #90, #157, #126, #92, and on other issues.

**Comment by qurle** [comment]
@GeekMagicClock, could you share the plans about open source code or reject it, please?

**Comment by luigimannoni** [comment]
I strongly second as well the request for publishing the source code @GeekMagicClock 

As a software engineer, I see potential in this hardware like I do as well for the rest of most of the AliExpress electronics/esp smartwatches/esp displays. 
There is a large community of developers like myself who possess extensive software expertise but lack the specialized hardware tinkering, soldering experience or 3d printers/tools to build a device like this from scratch. 

I personally love gadgets like the one you've built, and I always wanted to customize further for my needs. Please consider opening up the source code, even a partial release or a basic SDK to implement plugins would be a massive step forward.
